### PR TITLE
terraform 0.12.0

### LINKED ISSUE #13: terraform-provider-vra7 v0.2.0 not compatible with Terraform v0.12.0
**vRA 7.x version**
n/a
**Terraform version**
v0.12.0
**terraform-provider-vra7 plugin version**
v0.2.0

**Describe the bug**
After upgrade to Terraform 0.12, the vra7 provider can not be instantiated.

**To Reproduce**
1. upgrade to Terraform v0.12.0
2. `terraform plan`
3. `Error: Failed to instantiate provider "vra7" to obtain schema: Incompatible API version with plugin. Plugin version: 4,
Client versions: [5]`
4. `terraform init -upgrade`
5. 
```
Initializing the backend...

Initializing provider plugins...
- Checking for available provider plugins...

Provider "vra7" not available for installation.

A provider named "vra7" could not be found in the Terraform Registry.

This may result from mistyping the provider name, or the given provider may
be a third-party provider that cannot be installed automatically.

In the latter case, the plugin must be installed manually by locating and
downloading a suitable distribution package and placing the plugin's executable
file in the following directory:
    terraform.d/plugins/windows_amd64

Terraform detects necessary plugins by inspecting the configuration and state.
To view the provider versions requested by each module, run
"terraform providers".

Error: no provider exists with the given name
```
6. 
```
> terraform providers
.
└── provider.vra7
```

**Expected behavior**
The vra7 provider works with the upgraded Terraform application.

**Screenshots**
n/a
**Logs**
n/a

**Desktop (please complete the following information):**
 - OS: Windows 10

**Additional context**
Thanks!

### PR DESCRIPTION
**Pull request by skylerto**
This PR adds support for terraform 0.12.0.

The support was added following the guidelines provided by terraform (https://www.terraform.io/docs/extend/terraform-0.12-compatibility.html)

Most of what was done was for adding the vendor dependencies (go modules) as per the docs.

resolves #13

### CODE REVIEW

File: vra7/resource_vra7_deployment.go [hunk 1]
```diff
@@ -97,21 +97,13 @@ func resourceVra7Deployment() *schema.Resource {
 			"deployment_configuration": {
```
Comment (skylerto) [inline vra7/resource_vra7_deployment.go]: This was the only file pertaining to business logic that needed to be changed to support 12.

**Comment by markpeek** [comment]
@skylerto thank you for the PR to upgrade to 0.12. I've had those changes as well. But when using resource_specification I see this:
```
$ cat main.tf
provider "vra7" {
  username = var.vra_username
  password = var.vra_password
  tenant   = var.vra_tenant
  host     = var.vra_host
  insecure = true
}

//Component profile blueprint
resource "vra7_deployment" "deploy1" {
  count             = 1
  catalog_item_name = "CentOS 7.0 x64"
  wait_timeout      = 30

  resource_configuration = {
    Linux.cpu = "2"
  }

  reasons     = "Test"
  description = "deployment via terraform"
}
$ terraform plan

Error: Reference to undeclared resource

  on main.tf line 16, in resource "vra7_deployment" "deploy1":
  16:     Linux.cpu = "2"

A managed resource "Linux" "cpu" has not been declared in the root module.
```
I believe this has to do with the way 0.12 handles expressions differently from 0.11. Are you seeing this same thing with your changes?

**Comment by markpeek** [comment]
Ahhh, looking at the README changes in your PR I see you were going a similar path as I was, which was to quote the resource config value.
```
Linux.cpu => "Linux.cpu"
```
Likely all the examples will need to be changed. And verification of backwards compatibility with 0.11.

**Comment by skylerto** [comment]
@markpeek indeed, this was the only way I was able to get it working and passing vRA the correct values. I went down the route of trying to get the Maps working, but with terraform 12 a Map with a Elem value of Map is not valid.

ref `validateMapValues` in the terraform code: https://github.com/hashicorp/terraform/blob/7f5ffbfe9027c34c4ce1062a42b6e8d80b5504e0/helper/schema/schema.go#L1584-L1617

**Comment by skylerto** [comment]
@markpeek is there any next steps that I can take here to get this merged and released?

**Comment by markpeek** [comment]
@skylerto my apologies for the delay in getting this PR merged. Because this change is more than just updating the code for 0.12, we're being cautious. We are in the process of verifying/updating the unit and acceptance tests along with validating backwards compatibility with terraform 0.11. We will then work with HashiCorp on how best to roll out to our users. Thank you for your patience.

**Comment by skylerto** [comment]
Let me know how things go, would love to get this merged as opposed to managing our own release and fork.

Also have some other changes that we're working on that might be useful upstream.

**Comment by markpeek** [comment]
Thanks @skylerto! I'm going to follow up with a PR to get the acceptance tests working and then work with HashiCorp on the right way to release (versioning, etc.).

**Comment by skylerto** [comment]
@markpeek thanks! I'm by no means a vRA expert, I'm working with a client on an internal fork and figure I would contribute these back upstream; stay tuned from some more ideas. :)

**Comment by mcascone** [comment]
Hi all, please forgive my ignorance, but how do we pull down this new version? There's [no new release version since May 7.](https://github.com/terraform-providers/terraform-provider-vra7/releases) And currently, when i run `terraform init -upgrade` with Terraform v0.12, I still get the `no provider exists with the given name` error.
I'm sure i'm missing something dumb, but can anyone help?

**Comment by skylerto** [comment]
@mcascone that's a great question! There is no official release of this provider with terraform 12 support. There are options to build and use a copy of that artifact locally (which is what I do). If you have a look at HashiCorps documentation (https://www.terraform.io/docs/plugins/basics.html) there you will find where to place built providers. This one is as simple as running (`go build`) and placing it in the directory defined in that documentation. 

As for when the release will be available, that's a question for @markpeek.

**Comment by markpeek** [comment]
@mcascone and @skylerto currently working with HashiCorp on logistics to get a new plugin built to support v0.12. Release should be soon.

**Comment by skylerto** [comment]
Great, thanks for the update @markpeek!

**Comment by markpeek** [comment]
@mcascone @skylerto this is now installable from the HashiCorp provider repository. You may need to remove your old ones to get the new one (v0.3.0) installed. Note: we likely will update to the 0.12.6 sdk (per recent changes and advice) so there may be a v0.3.1 update out sometime soon.

**Pull request merged**